02.calendar/cal.rb

```diff
@@ -0,0 +1,254 @@
+require "date"
+require "optparse"
+
+# 数字表記の月：英語表記の月
+$month_name = {
+  "1" => "Jan", 
+  "2" => "Feb",
+  "3" => "Mar", 
+  "4" => "Apr", 
+  "5" => "May", 
+  "6" => "Jun", 
+  "7" => "Jul",
+  "8" => "Aug",
+  "9" => "Sep",
+  "10" => "Oct",
+  "11" => "Nov",
+  "12" => "Dec"
+}
+
+# 数字表記の曜日：英語表記の月
+$week_date = {
+  "0":"Sun", 
+  "1":"Mon", 
+  "2":"Tue", 
+  "3":"Wed",
+  "4":"Thr",
+  "5":"Fri",
+  "6":"Sat"
+}
+
+# ANSIエスケープシーケンス（背景：白、文字色：黒）
+ANSI_ESC = "\e[47m\e[30m"
+
+# ANSIエスケープシーケンスのクローズ
+ANSI_ESC_END = "\e[m"
+
+
+class Calendar
+
+ # 初期化
+ def initialize(year, month, day, flg_year, flg_month)
+  @year = year
+  @month = month
+  @day = day
+  @flg_year = flg_year
+  @flg_month = flg_month
+ end
+
+ # 月の最初の曜日を返す
+ def get_first_wday
+  return first_day = Date.new(@year, @month, 1).wday
+ end
+
+ # 月を数字表記から英語表記に直す
+ # 
+ # 引数
+ # month: 数字表記の月
+ #
+ # For examlpe:
+ #  convert_month_to_english("1")
+ #   => Jan
+ def convert_month_to_english(month)
+
+  # month_nameのハッシュデータのkeyを探索する
+  $month_name[month.to_s]
+
+ end
+
+ # カレンダーの曜日をコンソールに表示する
+ #
+ # For example:
+ #   print_all_weekdate
+ #      => Sun Mon Tue Wed Thr Fri Sat
+ def print_all_weekdate
+  $week_date.each_key do |key|
+   print $week_date[key] + " "
+  end
+  puts ""
```

Comment:
今のところRubyのhashは順序が保持されているので、`to_a`してjoinしてもいいかもしれませんね。
```suggestion
  puts $week_date.to_a.join(' ')
```

Reply:
あとはご自身でもおっしゃられている通り、そもそも定義の方を配列にしてしまえばさくっとjoinできますしその方がいいのではないでしょうか。現状だとhashであるメリットがなさそうです。